R/Data-class.R

```diff
@@ -290,95 +290,61 @@ DataMixture <- function(xshare = numeric(),
-                       "u entries must not be larger than Tmax")
-               ## (this has to be because the hazard
-               ## at Tmax is infinity -> all individuals stop
-               ## there)
-
-               o$check(all(object@u >= 0),
-                       "u entries must be non-negative")
-
-               ## DSB: why check not required?
-               #                                  o$check(all(object@t0 >= 0),
-               #                                          "t0 entries must be non-negative")
-               #
-               o$check(all(object@Tmax > 0),
-                       "DLT window needs to be greater than 0")
-
-               o$result()
-             })
-validObject(.DataDA())
-
-##' Initialization function for the `DataDA` class
-##'
-##' This is the function for initializing a `DataDA` class object.
-##'
-##' @param u the continuous vector of DLT free survival values
-##' @param Tmax the DLT observation period
-##' @param t0 time of initial dosing for each patient (default: 0)
-##' @param \dots additional parameters from \code{\link{Data}}
-##' @return the initialized \code{\linkS4class{DataDA}} object
-##'
-##' @export
-##' @keywords programming
-DataDA <- function(u=numeric(),
-                   Tmax=numeric(),
-                   t0=numeric(),
-                   ...)
-{
-
-  start <- Data(...)
+#' `DataDA`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' [`DataDA`] is a class for the time-to-DLT augmented data.
+#' It inherits from [`Data`] and it contains additional DLT free survival times.
+#'
+#' @note `survival time` here refers to the time period for which the subject
+#' did not experience any DLT, and is not referring to deaths.
+#'
+#' @slot u (`numeric`)\cr the continuous vector of DLT free survival times.
+#' @slot t0 (`numeric`)\cr time of initial dosing for each patient.
+#' @slot Tmax (`number`)\cr the DLT observation period.
+#'
+#' @aliases DataDA
+#' @export
+.DataDA <- setClass(
+  Class = "DataDA",
+  slots = c(
+    u = "numeric",
+    t0 = "numeric",
+    Tmax = "numeric"
+  ),
+  prototype = prototype(
+    u = numeric(),
+    t0 = numeric(),
+    Tmax = 0 + .Machine$double.xmin
+  ),
+  contains = "Data",
+  validity = validate_data_DA
+)
 
-  if(missing(t0))
-  {
-    t0 <- rep(0, length(u))
-  }
+# DataDA-constructor ----
 
-  .DataDA(start,
-          u = as.numeric(u),
-          t0 = t0,
-          Tmax = as.numeric(Tmax))
+#' @rdname DataDA-class
+#'
+#' @param u (`numeric`)\cr the continuous vector of DLT free survival times.
+#' @param t0 (`numeric`)\cr time of initial dosing for each patient.
+#'   Default to vector of 0s of length equal to length of `u`.
+#' @param Tmax (`number`)\cr the DLT observation period.
+#' @param ... parameters passed to [Data()].
+#'
+#' @export
+#' @example examples/Data-class-DataDA.R
+DataDA <- function(u = numeric(),
+                   t0 = numeric(length(u)),
+                   Tmax = 0 + .Machine$double.xmin,
```

Comment:
Hi @danielinteractive maybe I should just set `Tmax = 0` instead of `Tmax = 0 + .Machine$double.xmin` (and the same in corresponding unit tests and default constructor)? What do you think @danielinteractive? I added it because I though it is required that `Tmax` is strictly greater than `0`. But if this is the case, I would probably need to do this for `u` as well.

Reply:
@wwojciech hm good question. yeah I think it should be positive.